### PR TITLE
chore: clean up tests

### PR DESCRIPTION
**Pull request by enocom**
- Move tests closer to the related object (don't integration test everything)
- Introduce CloudSqlAdminApiTestDouble for better composable and intention revealing tests
- Prefer instance methods over static methods for better extensibility
- Rename FakeSslServer to what it fakes, i.e. "FakeCloudSqlProxyServer"
- Remove StubApiFetcherFactory and instantiate objects manually to set the stage for further object graph cleanup

### CODE REVIEW

**Review by shubha-rajan — DISMISSED** [review]

**Comment by enocom** [comment]
Yes, definitely a plan to restore test coverage. If you look at the deleted tests in CoreSocketFactory most of them are test low-level concerns (when does the SQL Admi API throw an error etc). I'm going to restore those tests at a lower level and keep the CoreSocketFactory about what it does.

File: core/src/test/java/com/google/cloud/sql/core/StubApiFetcherFactory.java
Comment (shubha-rajan) [inline core/src/test/java/com/google/cloud/sql/core/StubApiFetcherFactory.java:1]: This class may not be all that useful now, but once we introduce connectors and connectorBuilders, it'll be necessary, so maybe we shouldn't delete it

Comment (enocom) [reply core/src/test/java/com/google/cloud/sql/core/StubApiFetcherFactory.java:1]: Let's introduce it when we use it. I'm not convinced we want a factory.

**Comment by enocom** [comment]
I can make the changes more apparent and in smaller chunks. Will redo this.

**Pull request closed without merging**